#! /bin/sh /usr/share/dpatch/dpatch-run
## 07_tests_cmake_check_dependencies.dpatch by  <pgquiles@elpauer.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix CMake checks and linkage for tests

@DPATCH@

diff -rupdN witty-3.3.4+dfsg.orig/test/CMakeLists.txt witty-3.3.4+dfsg/test/CMakeLists.txt
--- witty-3.3.4+dfsg.orig/test/CMakeLists.txt	2014-11-17 11:55:38.000000000 +0100
+++ witty-3.3.4+dfsg/test/CMakeLists.txt	2015-04-07 19:26:37.951286000 +0200
@@ -1,3 +1,56 @@
+cmake_minimum_required(VERSION 2.4)
+
+IF(NOT MULTI_THREADED)
+  OPTION(MULTI_THREADED "Build multi-threaded httpd deamon (if possible)" ON)
+ENDIF(NOT MULTI_THREADED)
+
+INCLUDE( ${WT_SOURCE_DIR}/cmake/WtFindBoost.txt )
+INCLUDE(FindThreads)
+
+#IF(NOT BOOST_WT_FOUND)
+#  MESSAGE("** Error finding Wt required library: ")
+#  MESSAGE("** Could not find a boost installation in " ${BOOST_DIR} ".")
+#  MESSAGE("** It may be necessary to set appropriate values for the")
+#  MESSAGE("   variables BOOST_DIR, BOOST_COMPILER, and BOOST_VERSION")
+#  MESSAGE(FATAL_ERROR "** Wt requires the following C++ boost libraries:
+#  boost_date_time, boost_regex, and boost_program_options, boost_signals,
+#  and optionally boost_thread")
+#ENDIF(NOT BOOST_WT_FOUND)
+
+IF(BOOST_WT_MT_FOUND)
+  IF(MULTI_THREADED)
+    ADD_DEFINITIONS(-DTHREADED -D_REENTRANT -DBOOST_SPIRIT_THREADSAFE)
+    MESSAGE("** Enabling multi threading.")
+  ELSE(MULTI_THREADED)
+    MESSAGE("** Disabling multi threading.")
+    ADD_DEFINITIONS(-DBOOST_DISABLE_THREADS)
+  ENDIF(MULTI_THREADED)
+ELSE(BOOST_WT_MT_FOUND)
+  MESSAGE("** Disabling multi threading.")
+  ADD_DEFINITIONS(-DBOOST_DISABLE_THREADS)
+ENDIF(BOOST_WT_MT_FOUND)
+
+# Boost is used nearly everywhere, so we can put these here
+INCLUDE_DIRECTORIES(${BOOST_INCLUDE_DIRS})
+LINK_DIRECTORIES(${BOOST_LIB_DIRS})
+IF(WIN32)
+  IF(BOOST_DYNAMIC)
+    ADD_DEFINITIONS(-DBOOST_ALL_DYN_LINK)
+  ENDIF(BOOST_DYNAMIC)
+ENDIF(WIN32)
+
+IF(BOOST_WT_MT_FOUND)
+  IF(MULTI_THREADED)
+    ADD_DEFINITIONS(-DWT_THREADED -D_REENTRANT -DBOOST_SPIRIT_THREADSAFE)
+  ELSE(MULTI_THREADED)
+    ADD_DEFINITIONS(-DBOOST_DISABLE_THREADS)
+  ENDIF(MULTI_THREADED)
+ELSE(BOOST_WT_MT_FOUND)
+  ADD_DEFINITIONS(-DBOOST_DISABLE_THREADS)
+ENDIF(BOOST_WT_MT_FOUND)
+
+ADD_DEFINITIONS(-DWT_DEPRECATED_3_0_0)
+
 IF(ENABLE_LIBWTTEST)
   SET(TEST_SOURCES
     test.C
@@ -43,6 +96,10 @@ IF(ENABLE_LIBWTTEST)
 
   TARGET_LINK_LIBRARIES(test wt wttest ${BOOST_FS_LIB})
 
+  IF( UNIX )
+    TARGET_LINK_LIBRARIES( test dl )
+  ENDIF( UNIX )
+
   IF(ENABLE_LIBWTDBO)
     # Test all dbo backends
     SET(DBO_TEST_SOURCES
